packages/grafana-ui/src/components/Cascader/Cascader.tsx

```diff
@@ -82,31 +84,45 @@ export class Cascader extends React.PureComponent<CascaderProps, CascaderState>
       if (optionPath.indexOf(initValue) === optionPath.length - 1) {
         return {
           rcValue: optionPath,
-          activeLabel: option.label || '',
+          activeLabel: option.singleLabel || '',
         };
       }
     }
+    if (this.props.allowCustomValue) {
+      return { rcValue: [], activeLabel: initValue };
+    }
     return { rcValue: [], activeLabel: '' };
   }
 
   //For rc-cascader
   onChange = (value: string[], selectedOptions: CascaderOption[]) => {
     this.setState({
```

Comment:
Hmm, I wonder if we need to have the internal state. Can't we derive that from the Cascader's value (maybe prop)? This is a separate PR material

Reply:
It was needed when both the cascader and the select was controlled, by since I let go of the control of the select in this PR, it should be doable.